### DISCUSSION TITLE
How can I share code between adonis projects?

### DISCUSSION THREAD

**Discussion by equeraltowenea** [adonisjs/core#3762] [Help] [answered]
I have three projects that use the same models and I would like to use a Commons project with those models.

Example:

```
/my-projects/
	/common/
	/project-a/
	/project-b/
	/project-c/
```

I have solved this using Typescript references. The problem is that models use Adonis CORE and Lucid libraries and when doing the build it complains about not finding the libraries in IoC:

`error TS2307: Cannot find module '@ioc:Adonis/Lucid/Orm' or its corresponding type declarations.`

I understand that this happens because it is a library that does not start adonis and does not have the IoCs available

How can I fix this?

**Comment by equeraltowenea** [comment, marked as answer]
I have fixed it by adding types in tsconfig.

**Reply by kingflamez** [reply]
Please I'll appreciate an example
